builder/MyBuilder.pm

```diff
@@ -0,0 +1,60 @@
+package builder::MyBuilder;
+use strict;
+use warnings;
+use base 'Module::Build::XSUtil';
+
+use File::Spec::Functions qw(catfile catdir);
+use Config;
+
+sub new {
+    my ($class, %argv) = @_;
+    $class->SUPER::new(
+        %argv,
+        include_dirs => [catdir('msgpack-1.4.2', 'include')],
+        generate_ppport_h => catfile('lib', 'Data', 'MessagePack', 'ppport.h'),
+        cc_warnings => 1,
+    );
+}
+
+sub _set_mtime {
+    my $self = shift;
+    open my $fh, "<", "builder/mtime.txt" or die;
+    while (my $line = <$fh>) {
+        next if $line =~ /^#/ || $line !~ /\S/;
+        chomp $line;
+        my ($file, $mtime) = split /\t/, $line, 2;
+        $file = catfile(split /\//, $file);
+        die "miss $file" if !-e $file;
+        utime $mtime, $mtime, $file or die "utime $mtime, $mtime, $file: $!";
+    }
+}
+
+sub _build_msgpack {
+    my $self = shift;
+
+    $self->_set_mtime;
```

Comment:
We should restore original mtime in msgpack-1.4.2.tar.gz; otherwise end-users might be asked for having automake tools.